Fix readme (#46)
opentensor/async-substrate-interface: README.md

# Async Substrate Interface
This project provides an asynchronous interface for interacting with [Substrate](https://substrate.io/)-based blockchains. It is based on the [py-substrate-interface](https://github.com/polkascan/py-substrate-interface) project.

## Features

- Asynchronous API calls

> - Uses [bt-decode](https://github.com/opentensor/bt-decode) instead of [py-scale-codec](https://github.com/polkascan/py-scale-codec) for faster [SCALE](https://polkascan.github.io/py-scale-codec/) decoding.

Reviewer: ```suggestion
- Uses [bt-decode](https://github.com/opentensor/bt-decode) instead of [py-scale-codec](https://github.com/polkascan/py-scale-codec) for faster [SCALE](https://docs.substrate.io/reference/scale-codec/) decoding.
```

## Installation

To install the package, use the following command:

```bash
pip install async-substrate-interface
```

## Usage

Here is a basic example of how to use the async-substrate-interface:

```python
import asyncio
from async_substrate_interface import SubstrateInterface

async def main():

>     substrate = SubstrateInterface(

Reviewer 2: This won't work. `AsyncSubstrateInterface` is the `async` version, while `SubstrateInterface` is the sync version. 

Also, this doesn't use the async context manager (or `initialize`), so it won't work there either.

> from async_substrate_interface import SubstrateInterface
> 
> async def main():
>     substrate = SubstrateInterface(

Reviewer: ```suggestion
from async_substrate_interface import AsyncSubstrateInterface

async def main():
    substrate = AsyncSubstrateInterface(
```
Need to use the Async interface if we want to use `await`

        url="wss://rpc.polkadot.io"
    )

>     result = await substrate.query(
>         module='System',
>         storage_function='Account',
>         params=['5FHneW46xGXgs5mUiveU4sbTyGBzmto4oT9v5TFn5u4tZ7sY']
>     )

Reviewer: ```suggestion
      async with substrate as substrate:
        result = await substrate.query(
            module='System',
            storage_function='Account',
            params=['5CZs3T15Ky4jch1sUpSFwkUbYEnsCfe1WCY51fH3SPV6NFnf']
        )
```
Should use the context manager

Reviewer: old ss58 was not valid

    print(result)

asyncio.run(main())
```

## Contributing

Contributions are welcome! Please open an issue or submit a pull request.

## License

This project is licensed under the MIT License. See the [LICENSE](LICENSE) file for details.

## Contact

> For any questions or inquiries, please join the Bittensor Development Discord server: [Church of Rao](https://discord.gg/gavmT4R8sB).

Reviewer: ```suggestion
For any questions or inquiries, please join the Bittensor Development Discord server: [Church of Rao](https://discord.gg/XC7ucQmq2Q).
```

# Async Substrate Interface
This project provides an asynchronous interface for interacting with [Substrate](https://substrate.io/)-based blockchains. It is based on the [py-substrate-interface](https://github.com/polkascan/py-substrate-interface) project.

Additionally, this project uses [bt-decode](https://github.com/opentensor/bt-decode) instead of [py-scale-codec](https://github.com/polkascan/py-scale-codec) for faster [SCALE](https://docs.substrate.io/reference/scale-codec/) decoding.

## Installation

To install the package, use the following command:

```bash
pip install async-substrate-interface
```

## Usage

Here are examples of how to use the sync and async inferfaces:

```python
from async_substrate_interface import SubstrateInterface

def main():
    substrate = SubstrateInterface(
        url="wss://rpc.polkadot.io"
    )

    result = substrate.query(
        module='System',
        storage_function='Account',
        params=['5CZs3T15Ky4jch1sUpSFwkUbYEnsCfe1WCY51fH3SPV6NFnf']
    )

> 

Reviewer 2: add in a `substrate.close()` here, and you're good to go.

We want to hammer in the idea of intentionally closing the substrate object (either with the context manager or by directly calling `substrate.close()`)
